lib/src/async/concat.dart

```diff
@@ -71,7 +71,7 @@ class _ConcatStream<T> extends Stream<T> {
     }, onResume: () {
       if (currentSubscription != null) currentSubscription.resume();
     }, onCancel: () {
-      if (currentSubscription != null) return currentSubscription.cancel();
+      if (currentSubscription != null) currentSubscription.cancel();
```

Comment:
I think this is wrong. `onCancel` returns `void` or `Future`. Removing `return` will drop the future returned by `cancel()` on the floow. The proper fix is probably to `return null` in the `else` branch.

Reply:
Done.